cohttp-eio/src/server.ml

```diff
@@ -65,26 +60,44 @@ let internal_server_error_response =
 let bad_request_response =
   (Http.Response.make ~status:`Bad_request (), Body.Empty)
 
-let write_response (writer : Write.t)
-    ((response, body) : Http.Response.t * Body.t) =
+let write_response writer ((response, body) : Http.Response.t * Body.t) =
   let version = Http.Version.to_string response.version in
   let status = Http.Status.to_string response.status in
-  Write.string writer version;
-  Write.string writer " ";
-  Write.string writer status;
-  Write.string writer "\r\n";
-  Body.write_headers writer response.headers;
-  Write.string writer "\r\n";
-  match body with
-  | Fixed s -> Write.string writer s
-  | Chunked chunk_writer -> Body.write_chunked writer chunk_writer
-  | Custom f -> f writer
-  | Empty -> ()
+  Buf_write.string writer version;
+  Buf_write.char writer ' ';
+  Buf_write.string writer status;
+  Buf_write.string writer "\r\n";
+  Rwer.write_headers writer response.headers;
+  Buf_write.string writer "\r\n";
+  Body.write_body writer body
+
+(* request parsers *)
+
+let meth =
+  let open Eio.Buf_read.Syntax in
+  let+ meth = Rwer.(token <* space) in
+  Http.Method.of_string meth
+
+let resource =
+  let open Eio.Buf_read.Syntax in
+  Rwer.(take_while1 (fun c -> c != ' ') <* space)
+
+let[@warning "-3"] http_request t =
+  let open Eio.Buf_read.Syntax in
+  match Buf_read.at_end_of_input t with
+  | true -> Stdlib.raise_notrace End_of_file
+  | false ->
```

Comment:
Don't think this does anything.

Reply:
It needs to raise End_of_file. I have made changes to `take_while1` to throw this error. So now can remove this here. I think this should save some 'ns' in the request processing hot loop.

Thanks. Well spotted.